pinot-core/src/main/java/com/linkedin/pinot/core/segment/creator/impl/SegmentIndexCreationDriverImpl.java

```diff
@@ -424,6 +427,7 @@ private void handlePostCreation()
       throws Exception {
     final String timeColumn = config.getTimeColumnName();
     segmentName = config.getSegmentNameGenerator().getSegmentName(segmentStats.getColumnProfileFor(timeColumn));
+    updateSegmentStartEndTimeIfNecessary(segmentStats.getColumnProfileFor(timeColumn));
```

Comment:
timeColumn can be null or empty. Code needs to handle that

We check for != null in this file but reading code it timeColumn will not be null but empty if not specified